src/main/java/org/truffleruby/language/globals/IsDefinedGlobalVariableNode.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.language.globals;
+
+import org.truffleruby.language.RubyNode;
+import org.truffleruby.language.yield.YieldNode;
+
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+
+public abstract class IsDefinedGlobalVariableNode extends RubyNode {
+
+    private final String name;
+
+    public IsDefinedGlobalVariableNode(String name) {
+        this.name = name;
+    }
+
+    @Specialization(guards = "storage.isSimple()")
+    public Object executeDefined(
+            @Cached("getStorage()") GlobalVariableStorage storage) {
+        if (storage.isDefined()) {
+            return coreStrings().GLOBAL_VARIABLE.createInstance();
+        } else {
+            return nil();
+        }
+    }
+
+    @Specialization(guards = "storage.hasHooks()")
+    public Object executeDefinedHooks(VirtualFrame frame,
+            @Cached("getStorage()") GlobalVariableStorage storage,
+            @Cached("new()") YieldNode yieldNode) {
+        return yieldNode.dispatch(storage.getIsDefined());
+    }
+
+    protected GlobalVariableStorage getStorage() {
+        return getContext().getCoreLibrary().getGlobalVariables().getStorage(name);
+    }
+
+    public static IsDefinedGlobalVariableNode create(String name) {
+        return IsDefinedGlobalVariableNodeGen.create(name);
+    }
```

Comment:
The factory method (as well as other static methods) should be on top by convention with the rest of the code.